ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/IndexMappingTool.java

```diff
@@ -86,75 +88,80 @@ public Object parse(Object o) {
 
     @Override
     public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
-        @SuppressWarnings("unchecked")
-        List<String> indexList = parameters.containsKey("index")
-            ? gson.fromJson(parameters.get("index"), List.class)
-            : Collections.emptyList();
-        if (indexList.isEmpty()) {
-            @SuppressWarnings("unchecked")
-            T empty = (T) ("There were no results searching the index parameter [" + parameters.get("index") + "].");
-            listener.onResponse(empty);
-            return;
-        }
+        try {
+            List<String> indexList = new ArrayList<>();
+            if (StringUtils.isNotBlank(parameters.get("index"))) {
+                indexList = parameters.containsKey("index") ? gson.fromJson(parameters.get("index"), List.class) : Collections.emptyList();
+            }
 
-        final String[] indices = indexList.toArray(Strings.EMPTY_ARRAY);
+            if (indexList.isEmpty()) {
+                @SuppressWarnings("unchecked")
```

Comment:
are just trying to returnning a string on listener response? it's not recommended to supresswarning, try compose the string in a better way.

Reply:
but I don't have the context on this listener, when it's empty index list, should it return empty list or this string? 

try this?

```
if (indexList.isEmpty()) {
    String message = String.format("There were no results searching the index parameter [%s].", parameters.get("index"));
    listener.onResponse(createResponse(message));
    return;
}
```

Reply:
@mingshl , This is to return the correct type `T` as you can see line 89 `actionListener<T> listener`

The actionListener expects type `T` , not `String` type. I would suggest have a separate PR to enhance this part. No need to finish all issues in this PR.

Reply:
@pyek-bot , let's create issue tracking the issue and fix in new PR ?